package com.rodrigodev.xgen4j.test.organization.packages.packagesForErrorsAreGeneratedFromErrorNames_specialCases.error_name1;

import com.rodrigodev.xgen4j.test.organization.packages.packagesForErrorsAreGeneratedFromErrorNames_specialCases.RootNameException;
import com.rodrigodev.xgen4j.test.organization.packages.packagesForErrorsAreGeneratedFromErrorNames_specialCases.RootNameException;
import com.rodrigodev.xgen4j.test.organization.packages.packagesForErrorsAreGeneratedFromErrorNames_specialCases.ErrorCode;

/**
 * Autogenerated by XGen4J on January 1, 0001.
 */
public class ErrorName1Exception extends RootNameException {

    public static final ExceptionType TYPE = new ExceptionType();

    protected ErrorName1Exception(ErrorCode errorCode, String message) {
        super(errorCode, message);
    }

    protected ErrorName1Exception(ErrorCode errorCode, String message, Throwable cause) {
        super(errorCode, message, cause);
    }

    private static class ExceptionType extends RootNameException.ExceptionType {

        @Override
        protected RootNameException createException(ErrorCode errorCode, String message) {
            return new ErrorName1Exception(errorCode, message);
        }

        @Override
        protected RootNameException createException(ErrorCode errorCode, String message, Throwable cause) {
            return new ErrorName1Exception(errorCode, message, cause);
        }
    }
}
